src/convert-old-lara/convert.ts

```diff
@@ -0,0 +1,226 @@
+  embeddable: legacyEmbeddableType;
+}
+
+interface legacySection {}
+interface legacyPageType {
+  embeddable_display_mode: "stacked" | "carousel" | string;
+  text?: string;
+  is_completion: boolean;
+  is_hidden: boolean;
+  layout: string;
+  id: number;
+  name?: string | null;
+  position: number,
+  show_header?: boolean;
+  show_info_assessment: boolean;
+  show_interactive: boolean;
+  show_sidebar: boolean;
+  sidebar: string | null;
+  sidebar_title: string | null;
+  toggle_info_assessment: boolean;
+  additional_sections: legacySection;        // update when we support additional sections
+  embeddables: legacyEmbeddableWrapper[];
+}
+
+const getEmbeddablesArray = (embeddables: legacyEmbeddableType[], column: "primary" | "secondary" | null, embeddablesAreHidden: boolean): EmbeddableType[] => {
+  const embeddableArr:EmbeddableType[] = [];
+  embeddables.forEach((legacyEmbeddable: legacyEmbeddableType) => {
+    const isHalfWidth = !legacyEmbeddable.is_full_width;
+    delete legacyEmbeddable.is_full_width;
+    legacyEmbeddable.is_hidden = embeddablesAreHidden || legacyEmbeddable.is_hidden;
+    embeddableArr.push({column, is_half_width: isHalfWidth, ...legacyEmbeddable});
+  });
+  return embeddableArr;
+};
+
+const newSectionsResource = (resourcePage: legacyPageType): SectionType[] => { //have to use type any because of the change in type definition
+  const pageLayout = resourcePage.layout;
+  const sectionLayout = pageLayout === "l-full-width"
+                          ? "full-width"
+                          : pageLayout === "l-responsive"
+                            ? "responsive"
+                            : pageLayout;
+  const headerBlockEmbeddables: legacyEmbeddableType[] = [];
+  const primaryBlockEmbeddables: legacyEmbeddableType[] = [];
+  const secondaryBlockEmbeddables: legacyEmbeddableType[] = [];
+  const newSections: SectionType[] = [];
+  const headerBlockHidden = !resourcePage.show_header || false;
+  const primaryBlockHidden = !resourcePage.show_interactive;
+  const secondaryBlockHidden = !resourcePage.show_info_assessment;
+
+  resourcePage.embeddables.forEach((embeddableWrapper: any) => {
+    const section = embeddableWrapper.section;
+    switch (section) {
+      case "header_block":
+        headerBlockEmbeddables.push(embeddableWrapper.embeddable);
+        break;
+      case "interactive_box":
+        primaryBlockEmbeddables.push(embeddableWrapper.embeddable);
+        break;
+      case null:
+        secondaryBlockEmbeddables.push(embeddableWrapper.embeddable);
+        break;
+    }
+  });
+
+  const headerBlockSection = {
+    "layout": "full-width",
+    "is_hidden": false,
+    "secondary_column_collapsible": false,
+    "secondary_column_display_mode": resourcePage.embeddable_display_mode,
+    "embeddables": getEmbeddablesArray(headerBlockEmbeddables, null, headerBlockHidden)
+  };
+  const splitBlockSection = {
+    "layout": sectionLayout,
+    "is_hidden": false,
```

Comment:
If both `show_interactive` and `show_info_assessment` is false then it might be good to mark this splitBlockSection as hidden. That way if it is a better representation of the original content.